x-pack/filebeat/input/cometd/input_test.go

```diff
@@ -170,20 +168,22 @@ func TestInputStop_Wait(t *testing.T) {
 	require.Equal(t, 1, bay.GetConnectedCount())
 
 	var wg sync.WaitGroup
+	var waitForConnections sync.WaitGroup
 	wg.Add(1)
+	waitForConnections.Add(1)
 	go func() {
 		require.Equal(t, 1, bay.GetConnectedCount()) // current open channels count should be 1
 		event := <-eventsCh
 		assertEventMatches(t, expected, event) // wait for single event
 		wg.Done()
 		time.Sleep(100 * time.Millisecond)           // let input.Stop() be executed.
 		require.Equal(t, 0, bay.GetConnectedCount()) // current open channels count should be 0
+		waitForConnections.Done()
```

Comment:
Do you need a separate waitgroup for this? Can you just move the `wg.Done()` call above here and get the same result?

Reply:
Yes, I need that to wait for entire test to finishes. `wg.Done()` is to wait for event to get and `waitForConnections.Done()` is to see if all the connections are closed after `input.wait()` is called.

Reply:
Can you give the original `wg` variable a more descriptive name then, to make it clear why it needs to stay?

Reply:
Sure, maybe I can give it a name like `waitEventCollection`.

Reply:
or I guess `waitForEventCollection` would be better.